projects/storefrontlib/src/cms-components/product/product-list/container/product-list-component.service.ts

```diff
@@ -105,48 +114,48 @@ export class ProductListComponentService {
     };
   }
 
-  private getQueryFromRouteParams({
-    brandCode,
-    categoryCode,
+  /**
+   * Resolves the search query from the given `ProductListRouteParams`.
+   */
+  protected getQueryFromRouteParams({
     query,
+    categoryCode,
+    brandCode,
   }: ProductListRouteParams) {
     if (query) {
       return query;
     }
     if (categoryCode) {
       return this.RELEVANCE_ALLCATEGORIES + categoryCode;
     }
+
+    // TODO: drop support for brands as they should be treated
+    // similarly as any category.
     if (brandCode) {
       return this.RELEVANCE_ALLCATEGORIES + brandCode;
     }
   }
 
-  private search(criteria: SearchCriteria): void {
-    const query = criteria.query;
-    const searchConfig = this.getSearchConfig(criteria);
-
-    this.productSearchService.search(query, searchConfig);
-  }
-
-  private getSearchConfig(criteria: SearchCriteria): SearchConfig {
-    const result: SearchConfig = {
-      currentPage: criteria.currentPage,
-      pageSize: criteria.pageSize,
-      sortCode: criteria.sortCode,
-    };
-
-    // drop empty keys
-    Object.keys(result).forEach((key) => !result[key] && delete result[key]);
-
-    return result;
-  }
-
-  setQuery(query: string): void {
-    this.setQueryParams({ query, currentPage: undefined });
-  }
-
-  viewPage(pageNumber: number): void {
-    this.setQueryParams({ currentPage: pageNumber });
+  /**
+   * Performs a search based on the given search criteria.
+   *
+   * The search is delegated to the `ProductSearchService`.
+   */
+  protected search(criteria: SearchCriteria): void {
+    const currentPage = criteria.currentPage;
+    const pageSize = criteria.pageSize;
+    const sortCode = criteria.sortCode;
+
+    this.productSearchService.search(
+      criteria.query,
+      // TODO: consider dropping this complex passing of cleaned object
+      Object.assign(
+        {},
+        currentPage && { currentPage },
+        pageSize && { pageSize },
+        sortCode && { sortCode }
+      )
```

Comment:
Is there a reason for creating the search config in such a safe and complex way? Shouldn't the back-end drop the empty parameters?

Could just this work?
```ts
 this.productSearchService.search(
      criteria.query,
      { currentPage, pageSize, sortCode }
    );
```

Reply:
That's indeed why I added the TODO. I think it's overly complex, but there might be a good reason. There's at least failing tests which is why I kept is as-is.

Reply:
@Platonn you might have a saying here

Reply:
Ok, let's leave it for the next iteration then.